app/controllers/responses_controller.rb

```diff
@@ -19,6 +19,7 @@ def create
     @response            = Response.new(response_params)
     @response.survey     = @survey
     @response.ip_address = request.remote_ip
+    @response.language   = params[:locale] if I18n.available_locales.map(&:to_s).include?(params[:locale])
```

Comment:
Do we need the if bit?

Reply:
What if locale doesn't exist?

Reply:
Many thanks @spencerldixon I can rewrite this line.